rdflib_sqlalchemy/sql.py

```diff
@@ -62,31 +62,28 @@ def union_select(select_components, distinct=False, select_type=TRIPLE_SELECT):
                 cols = [c.subject, c.predicate, c.object]
             else:
                 raise ValueError('Unrecognized table type {}'.format(tableType))
-            select_clause = expression.select([functions.count().label('aCount')]).select_from(
-                expression.select(cols, whereClause).distinct().select_from(table))
+            select_clause = expression.select(*[functions.count().label('aCount')]).select_from(
+                expression.select(*cols).where(whereClause).distinct().select_from(table))
         elif select_type == CONTEXT_SELECT:
-            select_clause = expression.select([table.c.context], whereClause)
+            select_clause = expression.select(table.c.context)
+            if whereClause is not None:
+                select_clause = expression.select(table.c.context).where(whereClause)
         elif tableType in FULL_TRIPLE_PARTITIONS:
-            select_clause = table.select(whereClause)
+            select_clause = table.select().where(whereClause)
         elif tableType == ASSERTED_TYPE_PARTITION:
             select_clause = expression.select(
-                [table.c.id.label("id"),
+                *[table.c.id.label("id"),
                  table.c.member.label("subject"),
                  expression.literal(text_type(RDF.type)).label("predicate"),
                  table.c.klass.label("object"),
                  table.c.context.label("context"),
                  table.c.termComb.label("termcomb"),
                  expression.literal_column("NULL").label("objlanguage"),
-                 expression.literal_column("NULL").label("objdatatype")],
+                 expression.literal_column("NULL").label("objdatatype")]).where(
                 whereClause)
         elif tableType == ASSERTED_NON_TYPE_PARTITION:
-            select_clause = expression.select(
-                [c for c in table.columns] +
-                [expression.literal_column("NULL").label("objlanguage"),
-                 expression.literal_column("NULL").label("objdatatype")],
-                whereClause,
-                from_obj=[table])
-
+            all_table_columns = [c for c in table.columns] + [expression.literal_column("NULL").label("objlanguage"), expression.literal_column("NULL").label("objdatatype")]
+            select_clause = expression.select(*all_table_columns).select_from(table).where(whereClause)
```

Comment:
```
select_clause = expression.select(*all_table_columns).select_from(table)
if whereClause is not None:
    select_clause = select_clause.where(whereClause)
```
I had to hunt for a while but this is what is hit by that part of the test triples cases for all free variables in the query. I checked the compiled SQL by casting the `select_clause` to a string and I saw that because the `whereClause` is just `None` using SQLA's `where()` method yields a compiled `WHERE NULL` which blocks everything in the result set.

Reply:
Hi there. I've submitted the patch @jfahne outlined above targeted to your branch, @richardscholtens.

- https://github.com/richardscholtens/rdflib-sqlalchemy/pull/1

Reply:
@amotl, I believe I have added your two commits. However, I do not see any fix in your commits for the SQL problem. @jfahne did not provide an fix as far I can tell. He only states where the problem lies. Please correct me if I'm wrong. Maybe I missed something.

Reply:
I added a if else statement now and that seems to work for the PostgreSQL database for now:

```python
        elif tableType == ASSERTED_NON_TYPE_PARTITION:
            all_table_columns = [c for c in table.columns] + [expression.literal_column("NULL").label("objlanguage"), expression.literal_column("NULL").label("objdatatype")]
            if whereClause is not None:
                select_clause = expression.select(*all_table_columns).select_from(table).where(whereClause)
            else:
                select_clause = expression.select(*all_table_columns).select_from(table)
        **selects.append(select_clause)**

```